controllers/pkg/giteaclient/giteaclient.go

```diff
@@ -55,10 +55,12 @@ func GetClient(ctx context.Context, client resource.APIPatchingApplicator) (Gite
 	if client.Client == nil {
 		return nil, fmt.Errorf("failed creating gitea client, value of client.Client cannot be nil")
 	}
-
+	// check if instance is created
```

Comment:
you may want to mention "check-lock-check pattern implementation is followed" to make it easy for folks to search

Reply:
done.